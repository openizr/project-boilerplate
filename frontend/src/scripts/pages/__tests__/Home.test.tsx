/**
 * @jest-environment jsdom
 */

import * as React from 'react';
import Home from 'scripts/pages/Home';
import { act } from 'react-dom/test-utils';
import { render, unmountComponentAtNode } from 'react-dom';

let container = document.createElement('div');

describe('react/Home', () => {
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    jest.clearAllMocks();
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    ((container as unknown) as null) = null;
  });

  test('renders correctly - basic', () => {
    act(() => {
      render(<Home locale={{ LABEL_TEST: 'Test' }} />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
