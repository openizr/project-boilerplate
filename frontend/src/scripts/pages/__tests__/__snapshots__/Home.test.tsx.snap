// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Home renders correctly - basic 1`] = `
<p>
  <img
    alt="test"
    src="file"
  />
  <img
    alt="test"
    src="file"
  />
  Test
</p>
`;
